src/main/java/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,208 @@
+package internetshop.dao.jdbc;
+
+import internetshop.controller.LoginController;
+import internetshop.dao.OrderDao;
+import internetshop.dao.ShoppingCartDao;
+import internetshop.dao.UserDao;
+import internetshop.exception.DataProcessingException;
+import internetshop.lib.Dao;
+import internetshop.lib.Inject;
+import internetshop.model.Role;
+import internetshop.model.User;
+import internetshop.util.conection.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.log4j.Logger;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    private static final Logger LOGGER = Logger.getLogger(LoginController.class);
+
+    @Inject
+    OrderDao orderDao;
+    ShoppingCartDao shoppingCartDao;
+
+    @Override
+    public User create(User user) {
+        String query = "INSERT INTO internetshop.users (name, password) VALUES (?, ?)";
+        String query3 =
+                "SELECT internetshop.users.id "
+                        + "FROM internetshop.users WHERE name = ? AND password = ?";
+        String query2 = "INSERT INTO internetshop.users_roles VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, user.getName());
+            preparedStatement.setString(2, user.getPassword());
+            preparedStatement.executeUpdate();
+            PreparedStatement preparedStatement3 = connection.prepareStatement(query3);
+            preparedStatement3.setString(1, user.getName());
+            preparedStatement3.setString(2, user.getPassword());
+            ResultSet resultSet = preparedStatement3.executeQuery();
+            if (resultSet.next()) {
+                user.setId(resultSet.getLong("id"));
+                PreparedStatement preparedStatement2 = connection.prepareStatement(query2);
+                preparedStatement2.setLong(1, resultSet.getLong("id"));
+                preparedStatement2.setLong(2, 2L);
+                preparedStatement2.executeUpdate();
+            }
+            LOGGER.info("New user created");
```

Comment:
let's divide creating user and setting his roles by creating another method for it